### PR TITLE
fix(cli,api): omni agents create expose provider-agent-id / config-path / metadata flags (#372)

### LINKED ISSUE #372: omni agents create missing --provider-agent-id / --config-path / --metadata flags
## Problem

`omni agents create` exposes:
```
--name --provider --model --type --agent-provider
```

But does not expose any way to set:
- `provider_agent_id` (the provider's internal identifier — e.g. an agno agent name like `eugenia-seller`)
- `config_path`
- `metadata` (or specifically `metadata.providerAgentId`)

The dispatcher's `applyAgentFkOverrides` (`packages/api/src/plugins/agent-dispatcher.ts:3021`) reads:

```ts
const providerAgentId = (agentRow.metadata as Record<string, unknown>)?.providerAgentId ?? agentRow.configPath;
if (providerAgentId) effectiveInstance.agentInternalId = providerAgentId as string;
```

With neither field settable from the CLI, `agentInternalId` falls back to `"default"` and the call either hits the wrong agent or fails outright.

## Reproduction

```bash
omni agents create --name eugenia --provider <agno-provider-id> --model gpt-4 --type agent
omni instances update <instance-id> --agent-id <new-agent-id>
# Send a message
```

Dispatcher logs:
```
Dispatching to agent ... agentId: "default"
```

Then the provider returns 404/no-such-agent.

## Workaround

Direct DB update:
```sql
UPDATE agents
SET metadata = '{"providerAgentId": "eugenia-seller"}'::jsonb
WHERE id = '<agent-id>';
```

## Suggested fix

Add three flags to `omni agents create`:

```
--provider-agent-id <id>     The provider's internal agent identifier
--config-path <path>          Config file path for file-based providers
--metadata <json>             Free-form metadata as JSON string
```

The same gap should be addressed on `omni agents update` — see #360 (already tracking update flags).

## Source

`packages/cli/src/commands/agents.ts:130-180` (create command definition)

### PR DESCRIPTION
**Pull request by vasconceloscezar**
## Summary
- Add `--provider-agent-id`, `--config-path`, and `--metadata` flags to `omni agents create` so the provider-internal agent identifier can be set from the CLI
- Wire values through to the existing API fields (`metadata.providerAgentId`, `configPath`, `metadata`) that the dispatcher already reads
- Extract `buildCreateAgentBody` / `parseCreateMetadata` helpers to keep the action handler under the Biome cognitive-complexity limit

## Why
The dispatcher (`packages/api/src/plugins/agent-dispatcher.ts`) resolves `agentInternalId` from `agentRow.metadata?.providerAgentId ?? agentRow.configPath`. Without a CLI surface for these fields, `agentInternalId` fell back to `"default"` and provider dispatch (e.g. agno → `eugenia-seller`) failed. The API zod schema and SDK types already accepted `configPath` and `metadata`; this PR is the missing CLI wiring.

Precedence: if both `--metadata` (carrying its own `providerAgentId`) and `--provider-agent-id` are supplied, the flag wins and the rest of the metadata object is preserved.

Closes #372

## Test plan
- [x] `bun run typecheck` — all 19 packages pass
- [x] `bun test packages/cli` — 266 pass, 0 fail (4 new tests for #372)
- [x] `bun test packages/api/src/services/__tests__/agents.test.ts` — 13 pass, 0 fail
- [x] Pre-push full suite — 2986 pass, 0 fail
- [x] Biome lint — no new warnings introduced

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by vasconceloscezar** [comment]
# Review: FIX-FIRST — Missing `omni agents update` flags

## Phase 1: Spec Compliance — ❌ FAIL

**Criteria Status:**
- ✅ `omni agents create --provider-agent-id <id>` stores `metadata.providerAgentId` correctly
- ✅ `omni agents create --config-path <path>` stores `configPath` correctly  
- ✅ `omni agents create --metadata '{"key":"val"}'` stores arbitrary metadata correctly
- ❌ **`omni agents update` accepts the same three new flags — MISSING**

**Critical Gap:**
The WISH explicitly requires (lines 18, 41, 110-112):
- "Add `--provider-agent-id`, `--config-path`, `--metadata` flags to omni agents **create and update** (#372)"
- "`omni agents update` accepts the same three new flags"

The `update` subcommand (agents.ts:246-273) is missing all three flags AND the corresponding interface/function updates:
- No `--provider-agent-id`, `--config-path`, `--metadata` options in command definition
- UpdateAgentOptions interface (lines 22-30) missing these fields
- UpdateAgentBody interface (lines 32-39) missing these fields  
- buildUpdateAgentBody() (lines 125-148) doesn't map these fields
- Error message (line 261) doesn't mention them as updatable

## Phase 2: Code Quality — ✅ CREATE SIDE IS GOOD

What *is* implemented for `create`:
- ✅ JSON parsing validates syntax and object type with clear error messages (agents.ts:56-79)
- ✅ `--provider-agent-id` correctly overrides metadata.providerAgentId when both provided (agents.ts:75-76)
- ✅ All 4 new tests pass including edge cases (cli.test.ts:513-650)
- ✅ API schema already accepts fields (verified agents.ts:35,38,41)
- ✅ Mock API create handler accepts fields (mock-api.ts:165-167)

Issues:
- [MEDIUM] Mock API update handler doesn't handle configPath/metadata (mock-api.ts:195-204)

## Validation Results

```
✅ bun run typecheck — PASS
✅ bun test packages/cli/src/__tests__/cli.test.ts — 59 pass
✅ bunx biome check — No issues
```

## How to Fix

1. **Extend UpdateAgentOptions** (agents.ts:22-30):
   - Add `providerAgentId?: string; configPath?: string; metadata?: string;`

2. **Extend UpdateAgentBody** (agents.ts:32-39):
   - Add `configPath?: string; metadata?: Record<string, unknown>;`

3. **Extend buildUpdateAgentBody()** (agents.ts:125-148):
   - Reuse `parseCreateMetadata()` to handle JSON and precedence
   - Map both fields to the body

4. **Add flags to update subcommand** (agents.ts:248-256):
   ```
   .option('--provider-agent-id <id>', '...')
   .option('--config-path <path>', '...')
   .option('--metadata <json>', '...')
   ```

5. **Update error message** (agents.ts:261)

6. **Sync mock API** (mock-api.ts:195-204)

7. **Add tests for update** (cli.test.ts)

This is straightforward: `create` was done, `update` was missed. Mirror the create implementation.

File: packages/cli/src/commands/agents.ts [hunk 1]
```diff
@@ -37,6 +38,86 @@ interface UpdateAgentBody {
...
+interface CreateAgentOptions {
+  name: string;
+  provider: string;
+  model?: string;
+  type?: string;
+  agentProvider?: string;
+  providerAgentId?: string;
+  configPath?: string;
+  metadata?: string;
+}
```
[the remark this answers is not in the document]
Comment (vasconceloscezar) [reply packages/cli/src/commands/agents.ts:56]: Fixed in 72331a58. The update command now exposes --provider-agent-id, --config-path, and --metadata. Implementation mirrors create, except --metadata does a shallow fetch-then-merge so we never clobber unrelated metadata keys. --provider-agent-id still wins over --metadata providerAgentId. 7 new CLI tests cover invalid JSON, non-object JSON, config-path wiring, merge behavior, and flag precedence.
